transit_odp/pipelines/pipelines/dataset_etl/utils/dataframes.py

```diff
@@ -221,3 +226,21 @@ def get_min_date_or_none(dates):
 def get_max_date_or_none(dates):
     dates = [date for date in dates if date is not None]
     return max(dates) if dates else None
+
+
+def df_to_booking_arrangements(
+    revision: DatasetRevision, df: pd.DataFrame
+) -> Iterator[BookingArrangements]:
+    for record in df.reset_index().to_dict("records"):
+        service_code = record["service_code"]
+        service = Service.objects.get(
```

Comment:
As discussed - this would throw an exception if there are two files with the same service code and revision id. Lets merge the booking arrangements dataframe with service  dataframe obtained after loading services data. The obtain the service id to get the service instance which is passed in line 241